### PR TITLE
INDEXAPI-414 Allow adding new fields on updating objects in array

### PR DESCRIPTION
**Pull request by mpushkin**

### CODE REVIEW

**Review by terret — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -242,7 +242,7 @@ const main = {
           for (key in params.fieldsToUpdate.keySet()) {
             def value = params.fieldsToUpdate[key];
 
-            if (target[key] != null && target[key] != value) {
```
Comment (mpushkin) [inline src/index.ts:245]: @vladgolubev @terret if you remember context on why there was such a check, plz let me know :)

Comment (terret) [reply src/index.ts:245]: -

Comment (terret) [reply src/index.ts:245]: Thanks for the mention, I overlooked that.

Comment (terret) [reply src/index.ts:245]: It seems it must be with the old approach 
```target[key] != null```

Comment (mpushkin) [reply src/index.ts:245]: 1. here it says that null values are fine: https://www.elastic.co/guide/en/elasticsearch/reference/current/null-value.html

2. that's not what the check is doing as written :)
as currently written it allows update only if existing object in ES already has this field with a non-null value, and basically it doesn't allow adding new fields to existing objects

3. we could make the check that `value != null`, but then we won't be able to remove field values on updates, maybe we should make a script that removes field from object if value is set to null ?

Comment (mpushkin) [reply src/index.ts:245]: but yes, it looks like intentional behavior covered with tests:
https://github.com/shelfio/es-painless-fields/blob/e48a03219205d12074c5b1ea8fd4ab9705f2e7b9/src/es-tests/update-object-in-array.es.test.ts#L52-L66
no existing field -> no update
so let's not be hasty with this pr)

Comment (vladgolubev) [reply src/index.ts:245]: I don't remember. Check git blame, it might be from open source contributors

**Comment by mpushkin** [comment]
closing as unnecessary

**Pull request closed without merging**